src/hotspot/share/opto/regmask.hpp

```diff
@@ -68,24 +69,167 @@ class RegMask {
 
+  // Access word i in the register mask.
+  const uintptr_t& _rm_up(unsigned int i) const {
+    assert(orig_const || orig_ext_adr == &_RM_UP_EXT, "clone sanity check");
+    if (i < _RM_SIZE) {
+      return _RM_UP[i];
+    } else {
+      assert(_RM_UP_EXT != nullptr, "sanity");
+      assert( i >= _RM_SIZE, "sanity");
+      return _RM_UP_EXT[i - _RM_SIZE];
+    }
+  }
+
+  // Non-const version of the above.
+  uintptr_t& _rm_up(unsigned int i) {
+    assert(orig_ext_adr == &_RM_UP_EXT, "clone sanity check");
+    return const_cast<uintptr_t&>(const_cast<const RegMask*>(this)->_rm_up(i));
+  }
+
+  // The maximum word index
+  unsigned int _rm_max() const { return _rm_size - 1U; }
+
+  // Return a suitable arena for (extended) register mask allocation.
+  static Arena* _get_arena();
+
+  // Grow the register mask to ensure it can fit at least min_size words.
+  void _grow(unsigned int min_size, bool init = true) {
+    if(min_size > _rm_size) {
+      Arena* _arena = _get_arena();
+      min_size = round_up_power_of_2(min_size);
+      unsigned int old_size = _rm_size;
+      unsigned int old_ext_size = old_size - _RM_SIZE;
+      unsigned int new_ext_size = min_size - _RM_SIZE;
+      _rm_size = min_size;
+      if (_RM_UP_EXT == nullptr) {
+        assert(old_ext_size == 0, "sanity");
+        _RM_UP_EXT = NEW_ARENA_ARRAY(_arena, uintptr_t, new_ext_size);
+      } else {
+        assert(orig_ext_adr == &_RM_UP_EXT, "clone sanity check");
+        _RM_UP_EXT = REALLOC_ARENA_ARRAY(_arena, uintptr_t, _RM_UP_EXT,
+                                         old_ext_size, new_ext_size);
+      }
+      if (init) {
+        int fill = 0;
+        if(is_AllStack()) {
+          fill = 0xFF;
+          _hwm = _rm_max();
+        }
+        _set_range(old_size, fill, _rm_size - old_size);
+      }
+    }
+  }
+
+  // Make us a copy of src
+  void _copy(const RegMask& src) {
+    assert(_offset == src._offset, "offset mismatch");
+    _hwm = src._hwm;
+    _lwm = src._lwm;
+
+    // Copy base mask
+    memcpy(_RM_UP, src._RM_UP, sizeof(uintptr_t) * _RM_SIZE);
+    _all_stack = src._all_stack;
+
+    // Copy extension
+    if (src._RM_UP_EXT != nullptr) {
+      assert(src._rm_size > _RM_SIZE, "sanity");
+      _grow(src._rm_size, false);
+      assert(orig_ext_adr == &_RM_UP_EXT, "clone sanity check");
+      memcpy(_RM_UP_EXT, src._RM_UP_EXT,
+          sizeof(uintptr_t) * (src._rm_size - _RM_SIZE));
+    }
+
+    // If the source is smaller than us, we need to set the gap according to
+    // the sources all_stack flag.
+    if (src._rm_size < _rm_size ) {
+      int value = 0;
+      if (src.is_AllStack()) {
+        value = 0xFF;
+        _hwm = _rm_max();
+      }
+      _set_range(src._rm_size, value, _rm_size - src._rm_size);
+    }
+
+    assert(valid_watermarks(), "post-condition");
+  }
+
+  // Set a range of words in the register mask to a given value.
+  void _set_range(unsigned int start, int value, unsigned int range) {
```

Comment:
Suggestion: `length`, `words`, or similar would be a more idiomatic name for the third parameter (`range`), in my opinion.

Reply:
Thanks, updated.